src/frontend/state/authModalState.ts

```diff
@@ -0,0 +1,34 @@
+import { makeAutoObservable } from 'mobx'
+
+class AuthModalState {
+  private qaModeActive = false
+  private open = true
```

Comment:
nitpick but it might be good to use private fields instead of private keyword if you want to be sure this is not modified from outside the class at runtime.

This is what I've started to do with my new mobx classes

https://stackoverflow.com/questions/59641564/what-are-the-differences-between-the-private-keyword-and-private-fields-in-types
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

Reply:
Oh it seems there is a limitation on MobX https://mobx.js.org/observable-state.html#limitations 

> EcmaScript private fields (#field) are not supported. When using TypeScript, it is recommended to use the private modifier instead.

Reply:
ah gotcha. I guess that's only for observables but that could get confusing quick. cool. let's stick with the private keyword then